src/Personal.cpp

```diff
@@ -493,19 +493,23 @@ void CWorkers::NewDay() {
                 Workers[c].Happyness--;
             }
 
+            if (Workers[c].Happyness > 100) {
+                Workers[c].Happyness = 100;
+            }
+
             if (Workers[c].Happyness < -100) {
                 // Ihm reicht's! Er kündigt:
                 if (qPlayer.Owner == 0) {
-                    qPlayer.Messages.AddMessage(
-                        BERATERTYP_GIRL,
-                        bprintf(StandardTexte.GetS(TOKEN_ADVICE, 2000 + Workers[c].Typ + Workers[c].Geschlecht * 100), Workers[c].Name.c_str()));
+                    qPlayer.Messages.AddMessage(BERATERTYP_GIRL, bprintf(StandardTexte.GetS(TOKEN_ADVICE, 2000 + Workers[c].Typ + Workers[c].Geschlecht * 100),
+                                                                         Workers[c].Name.c_str()));
                 }
 
                 SLONG ExEmployer = Workers[c].Employer;
                 Workers[c].Employer = WORKER_RESERVE;
                 if (Workers[c].TimeInPool > 0) {
                     Workers[c].TimeInPool = 0;
                 }
+                Workers[c].Gehalt = Workers[c].OriginalGehalt;
```

Comment:
What's the purpose of resetting the salary?

Reply:
OriginalGehalt was also added by me back then. I use it to show with red/green when salary was reduced/increased compared to the original salary. 
When player A hires a guy, increases salary and then fires him again, this guy can appear in the hiring pool for player B. Then it would be weird if the salary already shows up in green. Therefore, I reset it to original value.